### ISSUE TITLE
[BUG] cannot edit longhorn replicas

### ISSUE THREAD

**Issue by koraykutanoglu** [longhorn/longhorn#3837]
## Describe the bug

When I install a pod in Kubernetes, when that pod needs a disk, longhorn cannot provide a PVC for that node.

## To Reproduce

There is no step for it. I am uninstalling an app directly in kubernetes and longhorn cannot provide PVC.

## Expected behavior

The Longhorn must be able to successfully supply PVC or properly bond the active existing PVCs to the pods.

## Environment

 - Longhorn version: V1.2.4
 - Installation method (e.g. Rancher Catalog App/Helm/Kubectl): Rancher
 - Kubernetes distro (e.g. RKE/K3s/EKS/OpenShift) and version: RKE
   - Number of management node in the cluster: 12
   - Number of worker node in the cluster: 9
 - Node config
   - OS type and version: Ubuntu Server 20.04 LTS
   - CPU per node: 8
   - Memory per node: 24
   - Disk type(e.g. SSD/NVMe): SSD
   - Network bandwidth between the nodes: 1G
 - Underlying Infrastructure (e.g. on AWS/GCE, EKS/GKE, VMWare/KVM, Baremetal): VMWare
 - Number of Longhorn volumes in the cluster: 32

## Additional context

The problem is actually simple. While using my Kubernetes environment without any problems, suddenly longhorn stopped being able to mount PVCs. So when I delete my active pods, the pod starts on another node, then longhorn cannot connect the existing PVC to my pod. Regarding this, it gives the following error:

```
MountVolume.MountDevice failed for volume "pvc-8afca2d4-4366-40bb-986b-5b132224b792" : rpc error: code = Internal desc = mount failed: exit status 32 Mounting command: mount Mounting arguments: -t ext4 -o defaults /dev/longhorn/pvc-8afca2d4-4366-40bb-986b-5b132224b792 /var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-8afca2d4-4366-40bb-986b-5b132224b792/globalmount Output: mount: /var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-8afca2d4-4366-40bb-986b-5b132224b792/globalmount: /dev/longhorn/pvc-8afca2d4-4366-40bb-986b-5b132224b792 already mounted or mount point busy.
```

Then I updated kubernetes and longhorn to latest version. However, the same situation continues. or when I want to build an app from scratch, longhorn can't provide any PVC.

```
MountVolume.MountDevice failed for volume "pvc-ecf939bc-1b64-4605-9bc7-f92cb904ee43" : rpc error: code = Internal desc = format of disk "/dev/longhorn/pvc-ecf939bc-1b64-4605-9bc7-f92cb904ee43" failed: type:("ext4") target:("/var/lib/kubelet/plugins/kubernetes.io/csi/pv/pvc-ecf939bc-1b64-4605-9bc7-f92cb904ee43/globalmount") options:("defaults") errcode:(exit status 1) output:(mke2fs 1.45.5 (07-Jan-2020) /dev/longhorn/pvc-ecf939bc-1b64-4605-9bc7-f92cb904ee43 is apparently in use by the system; will not make a filesystem here! )
```

In my investigations, if my pod is running on the 2nd machine and the replica is located on the 3rd machine and the 4th machine in longhorn at that time, longhorn cannot provide a replica for the pod that is lifted on the 2nd machine. I'm looking at the interface, the replica starts to form and is deleted again.

**Comment by shuo-wu** [comment]
- For the mount issue, can you enter the node host the volume is attached to and check the mounting list? In the latest version, Longhorn should be able to detect and resolve the **invalid/corrupted** mount point for the volumes.

- For the replica issue, do you mean you want a local replica for the volume/pod and the local replica cannot be launched? If YES, can you provide more info about it? Maybe there is no enough space on the 2nd node then Longhorn always fails at the scheduling.

**Comment by koraykutanoglu** [comment]
hello, this is very strange. There were no problems with our servers. My disk spaces were fine. I had enough space. However, the problem did not get fixed. We built the Kubernetes infrastructure and all the machines from scratch. The problem was resolved when we installed the latest version from scratch. However, we were never able to determine why these errors occurred. But one thing is doubtful. Before we had these problems, we had a problem with our disks in the data center. There was no active problem with our machines, but this must have affected the longhorn. This effect seemed like a lasting effect. It was a bad experience.
